pkg/resourceinterpreter/configurableinterpreter/luavm/lua_test.go

```diff
@@ -41,31 +41,55 @@ func TestGetReplicas(t *testing.T) {
 					Name:      "bar",
 				},
 				Spec: appsv1.DeploymentSpec{
-					Replicas: &replicas,
+					Replicas: pointer.Int32(1),
 					Template: corev1.PodTemplateSpec{
 						Spec: corev1.PodSpec{
+							NodeSelector: map[string]string{"foo": "foo1"},
+							Tolerations:  []corev1.Toleration{{Key: "bar", Operator: corev1.TolerationOpExists}},
 							Containers: []corev1.Container{
-								{},
+								{Resources: corev1.ResourceRequirements{Limits: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceCPU:    resource.MustParse("100m"),
+									corev1.ResourceMemory: resource.MustParse("100M"),
+								}}},
+								{Resources: corev1.ResourceRequirements{Limits: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceCPU:    resource.MustParse("1.2"),
+									corev1.ResourceMemory: resource.MustParse("1.2G"),
+								}}},
 							},
 						},
 					},
 				},
 			},
 			expected: true,
-			luaScript: ` function GetReplicas(desiredObj) 
-							nodeClaim = {} 
-							resourceRequest = {} 
-							result = {} 
-							replica = desiredObj.spec.replicas 
-							result.resourceRequest = desiredObj.spec.template.spec.containers[1].resources.limits 
-							nodeClaim.nodeSelector = desiredObj.spec.template.spec.nodeSelector 
-							nodeClaim.tolerations = desiredObj.spec.template.spec.tolerations 
-							result.nodeClaim = {} 
-							result.nodeClaim = nil
-							return replica, {} 
-						end`,
-			wantReplica:  1,
-			wantRequires: &workv1alpha2.ReplicaRequirements{},
+			luaScript: `
+local kube = require("kube")
+function GetReplicas(desiredObj)
+	replica = desiredObj.spec.replicas
+	requires = {
+		nodeClaim = {
+			nodeSelector = desiredObj.spec.template.spec.nodeSelector,
+			tolerations = desiredObj.spec.template.spec.tolerations
+    	},
+		resourceRequest = {},
+	}
+	for i = 1, #desiredObj.spec.template.spec.containers do
+		requires.resourceRequest.cpu = kube.resourceAdd(requires.resourceRequest.cpu, desiredObj.spec.template.spec.containers[i].resources.limits.cpu)
+		requires.resourceRequest.memory = kube.resourceAdd(requires.resourceRequest.memory, desiredObj.spec.template.spec.containers[i].resources.limits.memory)
+	end
```

Comment:
Can we have a higher abstract function/lib to get the requirements from a `pod template`?